extensions-contrib/opentsdb-emitter/src/main/java/io/druid/emitter/opentsdb/OpentsdbSender.java

```diff
@@ -110,12 +119,15 @@ private void sendEvents()
     public void run()
     {
       while (running) {
-        if (!eventQueue.isEmpty()) {
-          OpentsdbEvent event = eventQueue.poll();
+        try {
+          OpentsdbEvent event = eventQueue.take();
           events.add(event);
-          if (events.size() >= flushThreshold) {
-            sendEvents();
-          }
+        }
+        catch (InterruptedException e) {
+          log.error(e, "consumer take event failed!");
+        }
+        if (events.size() >= flushThreshold) {
+          sendEvents();
```

Comment:
The while loop still always be running! Problem have not been solved.

Reply:
`take` will block until an item becomes available, so I think it is okay. This function will spend most of its time waiting on `take` rather than in a spin loop.

Reply:
Sorry, you are right, but I do not think this is a elegant solution.